src/Nethermind/Nethermind.Merge.Plugin/MergePlugin.cs

```diff
@@ -262,16 +262,19 @@ public Task InitRpcModules()
             }
             Thread.Sleep(5000);
 
+            // Single block shouldn't take a full slot to run
+            // We can improve the blocks until requested, but the single block still needs to be run in a timely manner
+            double maxSingleImprovementTimePerSlot = _blocksConfig.SecondsPerSlot * 0.25;
```

Comment:
this should be configurable actually. for L2s, nethermind can use the whole slot time for building without issues.

Reply:
should be specified in BlocksConfig (the percentage of time out of the whole slot for block building)

Reply:
Made it confiurable